_security/configuration/opensearch-keystore.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: default
+title: OpenSearch keystore
+parent: Configuration
+nav_order: 50
+---
+
+# OpenSearch Keystore
+
+`opensearch-keystore` is a utility script for managing the OpenSearch keystore. This keystore provides a secure method to store sensitive information, such as passwords and keys, used in the OpenSearch cluster. The script allows users to create, list, add, and remove settings securely.
+
+## Usage
+
+```
+opensearch-keystore [command] [options]
+```
+
+### Commands
+
+- `create`: Initializes a new keystore. If a keystore already exists, this command will overwrite it.
+- `list`: Lists all settings in the keystore.
+- `add <setting-name>`: Adds a new setting to the keystore. Prompts for the value to be securely stored.
+- `add-file <file-name>`: Adds a new file to the keystore.
+- `remove <setting-name>`: Removes a setting from the keystore.
+- `upgrade <setting-name>`: Upgrades an existing setting within the keystore.
+- `passwd`: Sets a password for the keystore.
```

Comment:
IS there a specific permission required to access the keystore or just a password?

Reply:
You need permission to execute the `opensearch-keystore` script as well which requires access to the filesystem for where the script resides.

Reply:
How does a user download the script?

Reply:
Its included in the distribution. There's no need to download.